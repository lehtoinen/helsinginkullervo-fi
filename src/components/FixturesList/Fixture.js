import React from 'react';
import PropTypes from 'prop-types';

import styles from './Fixture.module.scss';

const Fixture = props => {
  const time = props.time.substr(0, props.time.length - 3);
  let winner;
  if (props.isCompleted) winner = props.homeScore - props.awayScore;

  return (
    <div
      className={`${styles.fixture} ${
        props.isCompleted ? styles.completed : null
      }`}
    >
      <div className={styles.subject}>
        <span className={styles.competition}> {props.competition}: </span>
        <span className={winner > 0 ? styles.winner : null}>
          {props.homeTeam} {props.homeScore}
        </span>{' '}
        -{' '}
        <span className={winner < 0 ? styles.winner : null}>
          {props.awayScore} {props.awayTeam}
        </span>
      </div>
      <div className={styles.info}>
        klo {time} <span className={styles.location}>{props.venue}</span>
      </div>
    </div>
  );
};

Fixture.propTypes = {
  time: PropTypes.string.isRequired,
  competition: PropTypes.string.isRequired,
  homeTeam: PropTypes.string.isRequired,
  awayTeam: PropTypes.string.isRequired,
  isCompleted: PropTypes.bool.isRequired,
  venue: PropTypes.string,
  homeScore: PropTypes.string,
  awayScore: PropTypes.string,
};

Fixture.defaultProps = {
  venue: null,
  homeScore: null,
  awayScore: null,
};

export default Fixture;
