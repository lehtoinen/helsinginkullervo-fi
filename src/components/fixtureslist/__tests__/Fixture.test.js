import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Fixture from '../Fixture';

configure({ adapter: new Adapter() });

describe('Fixture', () => {
  const component = ({
    fixture = {
      awayScore: '0',
      awayTeam: 'Kullervo/Überkleber',
      competition: 'Regions Cup',
      homeScore: '8',
      homeTeam: 'RiPS',
      isCompleted: true,
      time: '15:00:00',
      venue: 'Riihimäki Keskuskenttä TN',
    },
  }) => shallow(<Fixture {...fixture} />);

  test('renders', () => {
    expect(component({})).toMatchSnapshot();
  });

  test('renders with only required props defined', () => {
    expect(
      component({
        fixture: {
          awayTeam: 'Kullervo/Überkleber',
          competition: 'Regions Cup',
          homeTeam: 'RiPS',
          isCompleted: true,
          time: '15:00:00',
        },
      })
    ).toMatchSnapshot();
  });
});
