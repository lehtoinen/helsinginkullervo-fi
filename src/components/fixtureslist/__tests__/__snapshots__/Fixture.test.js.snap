// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixture renders 1`] = `
<div
  className="fixture completed"
>
  <div
    className="subject"
  >
    <span
      className="competition"
    >
       
      Regions Cup
      : 
    </span>
    <span
      className="winner"
    >
      RiPS
       
      8
    </span>
     
    -
     
    <span
      className={null}
    >
      0
       
      Kullervo/Überkleber
    </span>
  </div>
  <div
    className="info"
  >
    klo 
    15:00
     
    <span
      className="location"
    >
      Riihimäki Keskuskenttä TN
    </span>
  </div>
</div>
`;

exports[`Fixture renders with only required props defined 1`] = `
<div
  className="fixture completed"
>
  <div
    className="subject"
  >
    <span
      className="competition"
    >
       
      Regions Cup
      : 
    </span>
    <span
      className={null}
    >
      RiPS
       
    </span>
     
    -
     
    <span
      className={null}
    >
       
      Kullervo/Überkleber
    </span>
  </div>
  <div
    className="info"
  >
    klo 
    15:00
     
    <span
      className="location"
    />
  </div>
</div>
`;
